### ISSUE TITLE
Allow batching for StatefulConnection and ChannelWriter

### ISSUE THREAD

**Issue by mp911de** [mp911de/lettuce#456]
Allow sending `Collection<RedisCommand>` on `StatefulConnection` and `ChannelWriter`. Dispatching a collection of commands allows batching in batch sizes that are individual to each call.

Commands require partitioning regarding slots (Redis Cluster) and the write intention to hit the appropriate connection. In case of mixing read and write commands, these can hit different connections and get executed in a different order.

Executing commands one-by-one comes with an order although commands are executed on different connections. Redis' single-threaded nature serialized the execution order so a command order is rather predictable. 

Another approach could be falling back to write-intention if a batch contains commands with multiple intents. This way commands are executed in order but read commands would also use the write intent connection.

So basically there are two options:

1. Sacrifice order for using always the right `ReadFrom` setting
2. Sacrifice `ReadFrom` for retaining the execution order on a single node
